analyses/tumor-purity-exploration/02_explore-tumor-purity-thresholds.Rmd

```diff
@@ -0,0 +1,357 @@
+library(tidyverse)
+theme_set(theme_bw())
+```
+This notebook performs the following explores potential tumor purity thresholds that could be applied in other contexts to assess the influence of tumor purity on certain RNA-level analyses.
+
+## Functions
+
+This section defines some functions which are needed in this (and only for this!) notebook.
+
+```{r}
+# This is a helper function to calculated weighted means for tumor_fraction,
+# weighted across either broad_histology_display or cancer_group_display groups
+calc_weighted_mean <- function(df, group_var) {
+  df %>%
+    drop_na(tumor_fraction) %>%
+    add_count({{group_var}}) %>%
+    mutate(contribution = tumor_fraction * (n/sum(n))) %>%
+    summarize(weighted = sum(contribution)) %>% 
+    pull(weighted)
+}
+```
+
+
+
+
+```{r}
+# This function calculates the fraction of counts in a given group that would 
+# remain after a given universal threshold is applied
+calc_frac <- function(df, 
+                      threshold, 
+                      group_var) {
+  # establish total counts
+  total_count <- df %>%
+    count({{group_var}}, name = "total") 
+  
+  df %>%
+    filter(tumor_fraction <= threshold) %>%
+    # count after filtering
+    count({{group_var}}) %>%
+    # divide after/before
+    inner_join(total_count) %>%
+    mutate(frac_remaining = n/total) %>%
+    distinct()
+}
+
+
+# This function makes a barplot showing the remaining fraction of samples after 
+#  thresholding `mean_threshold` or `median_threshold`
+plot_remaining_frac <- function(df, 
+                                mean_threshold, 
+                                median_threshold, 
+                                group_var) {
+  
+  remaining_df <- bind_rows(
+    calc_frac(df, mean_threshold, {{group_var}}) %>%
+      mutate(threshold = "mean threshold"),
+    calc_frac(df, median_threshold, {{group_var}}) %>%
+      mutate(threshold = "median threshold")
+  )
+  
+  ggplot(remaining_df) + 
+    aes(x = {{group_var}}, 
+        y = frac_remaining, 
+        fill = threshold) + 
+    geom_col(color = "black", size = 0.5, 
+             position = position_dodge()) +
+    geom_hline(yintercept = 0.5) +
+    theme(axis.text.x = element_text(angle = 30, hjust = 1))
+}
+```
+
+
+```{r}
+# This function makes a barplot showing the remaining fraction of samples after 
+#   applying a per-group threshold. The inputted df is assumed to have been thresholded already
+plot_remaining_frac_group_threshold <- function(df, var, title, fill) {
+  ggplot(df, 
+         aes(x = {{var}}, y = percent_remaining)) + 
+    geom_col(fill = fill, color = "black" ) + 
+    geom_text(aes(y = percent_remaining+0.025, label = count_remaining), 
+              color = "red") +
+    labs(
+      y = "Percent of samples that remain after thresholding",
+      title = title, 
+      subtitle = "text labels are remaining counts"
+    ) +
+    theme(axis.text.x = element_text(angle = 30, size = 8, hjust = 1)) + 
+    ylim(c(0,1))
+}
+```
```

Comment:
Not something you have to change, but there's no real reason not to put all the functions in one chunk. You're not adding Markdown formatted text around them to add explanation, and you're not showing any output.